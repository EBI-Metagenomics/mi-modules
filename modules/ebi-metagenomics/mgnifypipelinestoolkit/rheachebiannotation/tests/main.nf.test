nextflow_process {

    name "Test Process MGNIFYPIPELINESTOOLKIT_RHEACHEBIANNOTATION"
    script "../main.nf"
    process "MGNIFYPIPELINESTOOLKIT_RHEACHEBIANNOTATION"

    tag "modules"
    tag "modules_ebimetagenomics"
    tag "mgnifypipelinestoolkit"
    tag "mgnifypipelinestoolkit/rheachebiannotation"

    test("add the rhea mapping based on the diamond uniref90 annotations") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ],
                    file(params.mgnify_pipelines_toolkit_testdata + 'diamond/proteins_short.faa', checkIfExists: true),
                    file(params.mgnify_pipelines_toolkit_testdata + 'diamond/diamond_short.tsv', checkIfExists: true)
                ]
                input[1] = file(params.mgnify_pipelines_toolkit_testdata + 'diamond/rhea2chebi_short.tsv', checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("add the rhea mapping based on the diamond uniref90 annotations - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ],
                    file(params.mgnify_pipelines_toolkit_testdata + 'diamond/proteins_short.faa', checkIfExists: true),
                    file(params.mgnify_pipelines_toolkit_testdata + 'diamond/diamond_short.tsv', checkIfExists: true)
                ]
                input[1] = file(params.mgnify_pipelines_toolkit_testdata + 'diamond/rhea2chebi_short.tsv', checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
