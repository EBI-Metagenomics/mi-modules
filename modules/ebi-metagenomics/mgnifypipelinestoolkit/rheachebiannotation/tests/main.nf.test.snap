{
    "add the rhea mapping based on the diamond uniref90 annotations - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,3142846dc664a6fabd2ae52a86143d1c"
                ],
                "rhea2proteins_tsv": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,3142846dc664a6fabd2ae52a86143d1c"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-01-27T21:07:34.219215"
    },
    "add the rhea mapping based on the diamond uniref90 annotations": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.tsv:md5,cbc725c9af469858b8ed48eb23ebbd40"
                    ]
                ],
                "1": [
                    "versions.yml:md5,3142846dc664a6fabd2ae52a86143d1c"
                ],
                "rhea2proteins_tsv": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.tsv:md5,cbc725c9af469858b8ed48eb23ebbd40"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,3142846dc664a6fabd2ae52a86143d1c"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-01-27T21:07:31.231072"
    }
}